gateway/rpc_storage_handler_test.go

```diff
@@ -0,0 +1,143 @@
+	case RevokeOauthRefreshToken:
+		// string is as= {the-token}=:{api-id}:oAuthRevokeToken
+		return fmt.Sprintf("%s:%s:oAuthRevokeRefreshToken", token, apiId)
+	case RevokeOauthRefreshHashedToken:
+		// string is as= {the-token}=:{api-id}:oAuthRevokeToken
+		return fmt.Sprintf("%s:%s:oAuthRevokeToken", token, apiId)
+	}
+	return ""
+}
+
+func getAccessToken(td tokenData) string {
+	return td.AccessToken
+}
+
+func getRefreshToken(td tokenData) string {
+	return td.RefreshToken
+}
+
+func TestProcessKeySpaceChangesForOauth(t *testing.T) {
+
+	cases := []struct {
+		TestName string
+		Event    string
+		Hashed   bool
+		GetToken func(td tokenData) string
+	}{
+		{
+			TestName: RevokeOauthToken,
+			Event:    RevokeOauthToken,
+			Hashed:   false,
+			GetToken: getAccessToken,
+		},
+		{
+			TestName: RevokeOauthHashedToken,
+			Event:    RevokeOauthHashedToken,
+			Hashed:   true,
+			GetToken: getAccessToken,
+		},
+		{
+			TestName: RevokeOauthRefreshToken,
+			Event:    RevokeOauthRefreshToken,
+			Hashed:   false,
+			GetToken: getRefreshToken,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.TestName, func(t *testing.T) {
+			ts := StartTest()
+			defer ts.Close()
+
+			globalConf := config.Global()
+			globalConf.HashKeys = tc.Hashed
+			config.SetGlobal(globalConf)
+
+			rpcListener := RPCStorageHandler{
+				KeyPrefix:        "rpc.listener.",
+				SuppressRegister: true,
+				HashKeys:         tc.Hashed,
+			}
+
+			myApi := loadTestOAuthSpec()
+			oauthClient := createTestOAuthClient(myApi, authClientID)
+			tokenData := getToken(t, ts)
+			token := tc.GetToken(tokenData)
+
+			var getKeyFromStore func(string) (string, error)
+			if tc.Event == RevokeOauthRefreshToken {
+				//Refresh token are threated in a different way due that they reside in a different level and we cannot access them directly
+				client := new(OAuthClient)
+				client.MetaData = oauthClient.MetaData
+				client.Description = oauthClient.Description
+				client.ClientSecret = oauthClient.GetSecret()
+				client.PolicyID = oauthClient.PolicyID
+				client.ClientRedirectURI = oauthClient.ClientRedirectURI
+
+				storage := myApi.OAuthManager.OsinServer.Storage
+				ret := &osin.AccessData{
+					AccessToken:  tokenData.AccessToken,
+					RefreshToken: tokenData.RefreshToken,
+					Client:       client,
+				}
+				storage.SaveAccess(ret)
+
+				getKeyFromStore = func(token string) (string, error) {
+					accessData, err := storage.LoadRefresh(token)
+					var refresh string
+					if accessData != nil {
+						refresh = accessData.RefreshToken
+					}
+					return refresh, err
+				}
+			} else {
+				getKeyFromStore = GlobalSessionManager.Store().GetKey
+				GlobalSessionManager.Store().DeleteAllKeys()
+				GlobalSessionManager.Store().SetKey(token, token, 100)
+				_, err := GlobalSessionManager.Store().GetKey(token)
+				if err != nil {
+					t.Error("Key should be pre-loaded in store previously so the test can perform the revoke action.")
```

Comment:
Why not just `t.Fatal` here? We need the key to be pre loaded I assume, so its safe to just exit here as the rest of the test will generate errors too.

Reply:
makes sense, done.